WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -1,6 +1,7 @@
 class CompetitionsController < ApplicationController
   before_action :authenticate_user!, except: [:show]
   before_action :can_admin_results_only, only: [:index, :post_announcement, :post_results, :admin_edit]
+  before_action :can_create_competition_only, only: [:new]
```

Comment:
Hmm... this looks identical to the `can_create_competition_only` down on line 23. If you remove this line, does your test still pass?